SalesforceFoundation/HEDAP
# Creating Objects/Fields for "Time Block" and "Course Offering Schedule"

# Critical Changes

# Changes
## New Time Block Object
- We've added a new Time Block object that you can use to track blocks of time that are relevant to your educational institution. For example:
  - A university might use Time Blocks to define the hours when specific courses are offered.
  - A K-12 school might use them to track class periods.
- For new HEDA installations, the Time Block related list is automatically added to the Account page layout. If your organization has an existing HEDA installation, you'll want to manually add the Time Block related list to your Account page layouts to best take advantage of this new object.

## New Course Offering Schedules Junction Object
- We've added a new Course Offering Schedules junction object so you can associate one or more schedules with a given Course Offering. You can also associate a Time Block with a Course Offering Schedule. For example, a Course Offering called "BIO 101 - Fall 2019" might have multiple Course Offering Schedules listed to reflect that the course is offered on different days of the week, at different locations, and during different blocks of time.
- For new HEDA installations, the Course Offering Schedules related list is automatically added to the Course Offering page layout. If your organization has an existing HEDA installation, you'll want to manually add the Course Offering Schedules related list to your Course Offering page layouts to best take advantage of this new object.

# Issues Closed

#732

# New Metadata
- Time Block (Time_Block__c.object)
- Time_Block__c-EDA Time Block Layout.layout
- Course Offering Schedule (Course_Offering_Schedule__c.object)
- Course_Offering_Schedule__c-EDA Course Offering Schedule Layout.layout
- Time_Block__c-ca.objectTranslation
- Time_Block__c-en_US.objectTranslation
- Time_Block__c-es.objectTranslation
- Time_Block__c-fr.objectTranslation
- Course_Offering_Schedule__c-ca.objectTranslation
- Course_Offering_Schedule__c-en_US.objectTranslation
- Course_Offering_Schedule__c-es.objectTranslation
- Course_Offering_Schedule__c-fr.objectTranslation



# Deleted Metadata

# Testing Notes
- Ensure Time Block related list is on your Account page layout's related lists
- Ensure Course Offering Schedule related list to your Course Offering's related lists
- Create an Educational Institution Account
- Create one or more Time Blocks for the account
- Create a Course Offering and select a Time Block for it to look up to
- From the Course Offering's Related lists, add a Course Offering Schedule
